### PR TITLE
Use class+support node 12+

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
## Updates

- Support Node.js 12+
- Convert encodings to class
- Remove `statuses`
- Minify the conditions
- Update deps.

Related: #135

### CODE REVIEW

**Comment by jonathanong** [comment]
looks good! thanks

File: .eslintrc [hunk 1]
```diff
@@ -1,2 +1,3 @@
 ---
 extends: standard
+parser:  "babel-eslint"
```
Comment (jonathanong) [inline .eslintrc:3]: why do we need this? seems to work fine without it for me

Comment (jonathanong) [reply .eslintrc:3]: is one of the class features not supported by regular eslint right now?

Comment (3imed-jaberi) [reply .eslintrc]: I added `eslint-babel` to support the static attribute. and I think the class syntax is much cleaner.
I fixed this now by remove the static attribute and work with old way (static).

```js
MyClass.MyStaticAttribute = "MyValue"
```

Comment (jonathanong) [reply .eslintrc]: interesting that eslint still doesn't support it...

**Review by jonathanong — APPROVED** [review]

**Pull request merged**